File: subprogramas/linguagem C/amb_ref_locais.c

// Ambientes em referenciais locais: variáveis estáticas e dinâmicas
// Em C, temos ambos os exemplos, onde normalmente seu padrão é a chamada stack-dinâmica, onde por padrão ela  é estática, mas por meio de subprogramas (funções) ou de sentenças propostas pelo código pode ser alterada (tornando-se dinâmica).

#include <stdio.h>
#include <stdlib.h>

int somador (int array[], int array_tam)
{
    static int soma = 0; // --> aqui é onde a variável se torna ESTÁTICA
    
}
    

File: subprogramas/linguagem PHP/amb_ref_locais.php

<?php

function somador($array) {
    static $soma = 0; // --> aqui é onde a variável se torna ESTÁTICA
    $var_a = 0; // --> variável definida por padrão como dinâmica (em PHP todas as variáveis são dinâmicas por padrão)
    
    foreach ($array as $value) {
        $soma += $value;
    }
    
    return $soma;
}

// Exemplo de uso
$array = [1, 2, 3, 4, 5];
$soma_total = somador($array);
echo "A soma dos elementos é: $soma_total\n";
?>
